### PR TITLE
(WIP) NETKVM Bringup

### PR DESCRIPTION
**Pull request by BenNottelling**
## Purpose

Import the redhat netkvm driver into ReactOS

JIRA issue: [CORE-15841](https://jira.reactos.org/browse/CORE-15841)

## TODO

- Fix compiler errors 
- Address reviews 
- Test driver functionality

### CODE REVIEW

**Comment by BenNottelling** [comment]
@HBelusca Like this? https://github.com/reactos/reactos/pull/1561/commits/447e1139edf02bcaaee907cbdbf8a1e5e2d68cda

File: drivers/network/dd/netkvm/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    wxp/ParaNdis5-Oid.c)
+
+include_directories(${REACTOS_SOURCE_DIR}/sdk/include/reactos/libs/VirtIO)
+include_directories(${REACTOS_SOURCE_DIR}/sdk/include/reactos/libs/VirtIO/Tools)
+add_library(netkvm MODULE ${SOURCE} wxp/parandis.rc)
+set_module_type(netkvm kernelmodedriver)
+add_importlibs(netkvm ndis ntoskrnl hal)
+add_cd_file(TARGET netkvm DESTINATION reactos/system32/drivers FOR all)
+if(NOT MSVC)
+  add_target_compile_flags(logevent "-Wno-unused-but-set-variable")
```
Comment (learn-more) [inline drivers/network/dd/netkvm/CMakeLists.txt]: logevent?

Comment (BenNottelling) [reply drivers/network/dd/netkvm/CMakeLists.txt]: That's what was here https://git.reactos.org/?p=reactos.git;a=blob;f=modules/rosapps/applications/sysutils/logevent/CMakeLists.txt;hb=ad80715b1adb444a514b43a7c6c868b0cac4be34#l4
*shrugs*

Comment (learn-more) [reply drivers/network/dd/netkvm/CMakeLists.txt]: because it is adding a compile flag to the target `logevent`

Comment (binarymaster) [reply drivers/network/dd/netkvm/CMakeLists.txt]:
```suggestion
  add_target_compile_flags(netkvm "-Wno-unused-but-set-variable")
```

**Comment by BenNottelling** [comment]
Whoops, this got closed when I rebased everything and started with clean code.
FYI, code almost compiles in MSVC, but I need help fixing the last few bugs

File: sdk/include/ddk/ndis.h [hunk 2]
```diff
@@ -4296,7 +4296,7 @@ NdisDprReleaseSpinLock(
 
 /*
  * VOID
- * NdisReadRegisterUshort(
+ * NdisWtiteRegisterUshort(
```
Comment (ThFabba) [inline sdk/include/ddk/ndis.h]: Replace one typo with another? :p

Comment (BenNottelling) [reply sdk/include/ddk/ndis.h]: Well, it's closer now :P 
I'll fix that in the next commit

**Comment by learn-more** [comment]
@BenNottelling Please specify with what you need help :)

**Comment by BenNottelling** [comment]
@learn-more Hey Mark, currently we need WDF headers for compiling the driver, which is where PR #1734 comes in, but it's not the actual headers needed for it, I was kinda planing on building out the whole header collection

**Comment by Mouvedia** [comment]
Should this be closed in favor of #2413?

**Comment by binarymaster** [comment]
Superseded by #2413 (got merged), thank you for your work!

**Pull request closed without merging**